gcore/gdaldriver.cpp

```diff
@@ -949,16 +960,56 @@ GDALDataset *GDALDriver::CreateCopy( const char * pszFilename,
     if( pfnProgress == nullptr )
         pfnProgress = GDALDummyProgress;
 
+/* -------------------------------------------------------------------- */
+/*      If no INTERLEAVE creation option is given, we will try to add   */
+/*      one that matches the current srcDS interleaving                 */
+/* -------------------------------------------------------------------- */
+    char** papszOptionsSmart = 0;
+    const bool bHasInterleaveCreationOptions =
+         (CSLFetchNameValue(papszOptions, "INTERLEAVE") != 0);
+    if (!bHasInterleaveCreationOptions && poSrcDS)
+    {
+        const char* srcInterleave = poSrcDS->GetMetadataItem("INTERLEAVE", "IMAGE_STRUCTURE");
+        const bool isENVIDriver = !strcmpi(this->GetDescription(), "ENVI");
+        char** dstInterleaveOptions = (char**)CPLCalloc(1, sizeof(char*));
+        dstInterleaveOptions = CSLSetNameValue(dstInterleaveOptions, "INTERLEAVE",
+            !srcInterleave ?
+              isENVIDriver ? "BSQ" : "BAND" :
+            !isENVIDriver ? srcInterleave :
+            !strcmpi(srcInterleave, "BAND") ? "BSQ" :
+            !strcmpi(srcInterleave, "PIXEL") ? "BIP" :
+            !strcmpi(srcInterleave, "LINE") ? "BIL" :
+            srcInterleave);
+        const bool bIsDstInterleaveSupported = (GDALValidateCreationOptions(
```

Comment:
the issue with GDALValidateCreationOptions() is that it will emit warnings in the standard error if the output driver doesn't support the INTERLEAVE creation options, or with values diffferent than the ones we provide. My above suggestion would avoid this.